homeassistant/components/api/__init__.py

```diff
@@ -387,6 +387,25 @@ async def post(
             )
 
         context = self.context(request)
+
+        if response_requested := "return_response" in request.query:
```

Comment:
Until `hass.services.supports_response` has been improved, let's check the service exists
```suggestion
        if not hass.services.has_service(domain, service):
            raise HTTPBadRequest from ServiceNotFound(domain, service)

        if response_requested := "return_response" in request.query:
```